### DISCUSSION TITLE
Message Group Id is expected as string

### DISCUSSION THREAD

**Discussion by agilan-selvam-zoomrx** [aws/aws-sdk-php#2876] [Q&A] [answered]
Hey Everyone,

I have recently upgraded my aws/aws-sdk-php library from 3.271.9 to 2.298.1 while upgrading my Laravel from 9.x to 10.x. After this upgrade, while pushing events to sqs, it expected message group id as string type. Earlier, we send it as integer. But now, it has been throwing error.

I could see the file which included in the upgrade QueryCompatibleInputMiddleware.php is throwing this error. Can someone clarify, whether it is a bug or should we change message group id to string?

Thanks in advance.

**Comment by yenfryherrerafeliz** [comment, marked as answer]
Hi @Agilan259, this behavior is indeed expected. What happened was that we were not validating the input before, but since SQS migrated from XML to JSON protocol, we now need to make sure the correct type is provided to prevent service side errors. You can check [here](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html#API_SendMessage_RequestSyntax) the parameter types for a send-message input, where you will see that MessageGroupId is string.

Please let me know if you have any questions.

Thanks!
